### ISSUE TITLE
Incorrect PATH in integrated terminal

### ISSUE THREAD

**Issue by hameerabbasi** [Microsoft/vscode#69954]
- VSCode Version: 1.31.1
- OS Version: macOS 10.14.4 beta 4

Steps to Reproduce:

1. Install ZSH and Anaconda

Set `.zshrc` to (substitute your Anaconda path):
```
. $HOME/anaconda/etc/profile.d/conda.sh
export PATH="$PATH:$HOME/anaconda/bin"
```

2. Launch zsh in either iTerm or the macOS terminal and do `echo $PATH`.

You get (example)
```
/Users/hameerabbasi/anaconda/condabin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Wireshark.app/Contents/MacOS:/Users/hameerabbasi/anaconda/bin
```

3. Launch zsh in the VSCode integrated terminal and do the same. Output is:
```
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Wireshark.app/Contents/MacOS:/usr/local/sbin:/Users/hameerabbasi/anaconda/bin:/Users/hameerabbasi/anaconda/condabin:/Users/hameerabbasi/anaconda/bin
```

The upshot of this is that we cannot use the conda-installed Python within zsh in the integrated terminal. I've reported it here because the Python extension is disabled, just the integrated terminal.

Does this issue occur when all extensions are disabled?: Yes/~~No~~

**Comment by hameerabbasi** [comment]
It seems it happens in `bash` as well, so not specific to `zsh`.

**Comment by jarlefosen** [comment]
It sounds like the VSCode instance is running `bash` instead of `zsh`, causing it to not source the `.zshrc` file as you expect.

Can you try and change your MacOS default shell to be `zsh` by following this guide?
http://osxdaily.com/2012/03/21/change-shell-mac-os-x/

Please note that the path to the `zsh` binary may be different on your machine based on how you installed it. For me I set it to be `/usr/local/bin/zsh`

_An alternative solution_ is to create a common shell startup script that you can source from both `.zshrc` and `.bash_profile` so you get the same `$PATH` in both `zsh` and `bash`.

**Comment by hameerabbasi** [comment]
That fixes it. That's kind of weird because I used `chsh -s /usr/local/bin/zsh` to change it... And VSCode would launch it, but the PATH was still weird.

I had the lines in both `.bashrc` and `.zshrc` and tested with both `zsh` and `bash`, and the issue was the same. But the terminal prefs section fixed it.

**Comment by rikvanderkemp** [comment]
This happens on Linux as well. Whenever I open a project and go straight to the terminal this happens:

![deepinscreenshot_select-area_20190307174627](https://user-images.githubusercontent.com/347308/53973436-08356300-4101-11e9-9d94-21e6b45ff3a3.png)

Running fish, version 3.0.1

My default shell is set to fish

**Comment by hameerabbasi** [comment]
Okay, it happens again after a reboot on both shells.

**Comment by Tyriar** [comment]
When VS Code is launched from finder on mac VS Code runs `$SHELL -ilc` in order to get a reasonable development environment:

https://github.com/Microsoft/vscode/blob/ef32b4594599366dcf33afbf495c9609b923b983/src/vs/code/node/shellEnv.ts#L24-L28

When you launch the terminal that will again run your terminal start up scripts (which ones depend on the value in `terminal.integrated.shellArgs.osx`). This can mangle your PATH if your scripts aren't idempotent. I think the problems you're experiencing are by design currently as there isn't really an elegant solution to the problem.

**Comment by hameerabbasi** [comment]
The issue here is that it completely mangles the ability of `conda` to work properly in Python, so `conda activate <env>` doesn't work. It kills the development workflow to have a different shell, especially since I use live-share to code with a co-worker.

I'm not sure what you mean by "idempotent".

**Comment by rikvanderkemp** [comment]
@Tyriar this doesn't only happen on MacOs. It happens on Linux as well. I wouldn't mind providing more information if I know what you need. This behavior only started recently.

**Comment by hameerabbasi** [comment]
Neither would I. I'd love to help debug.

**Comment by hameerabbasi** [comment]
> This behavior only started recently.

Maybe this is unrelated to my switch to ZSH, because as I stated, it happens on bash as well, with the old configuration. It probably just was a coincidence that it happened at the same time.

**Comment by rikvanderkemp** [comment]
@hameerabbasi Just noticed that when I set the integrated terminal to bash it seems to work. Would you mind trying this? Maybe we can narrow down what is going on:

```
"terminal.integrated.shell.linux": "/usr/bin/bash",
```

**Comment by hameerabbasi** [comment]
```
/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Wireshark.app/Contents/MacOS:/Users/hameerabbasi/anaconda/condabin:/Users/hameerabbasi/bin:/Users/hameerabbasi/anaconda/bin:/Users/hameerabbasi/.local/bin:/Users/hameerabbasi/anaconda/bin
```

The path `/Users/hameerabbasi/anaconda/condabin` should be at the start.

**Comment by rikvanderkemp** [comment]
@Tyriar I seem to may have found the issue for Fish Shell (3.0.1). Whatever I set the cwd to, it will always go to my home directory:

Using:
```
"terminal.integrated.cwd" : "some_random_path"
```

If I switch to bash, the shell navigates properly on new terminals.

**Comment by hameerabbasi** [comment]
This happens for me in bash as well... `cwd` set to `.`,
```
$ echo $PWD
/Users/hameerabbasi
```

**Comment by rikvanderkemp** [comment]
That's really odd @hameerabbasi 

I have confirmed that, for me, downgrading to Fish version 3.0.0 resolves my issues. So this is most likely a bug in Fish 3.0.1-1. 

_However_, I find it odd that your bash does the same thing. Maybe this is due to having a different bash version? I'm running 5.0.0(1)-release. None the less, it's an odd behaviour

Maybe try again, but this time using /bin/sh ?

**Comment by hameerabbasi** [comment]
Hmm, I'm not sure... The `pwd` issues are with `zsh` again, but the `$PATH` issues happen everywhere.

**Comment by hameerabbasi** [comment]
Something else is going on here that I don't understand, maybe some cache was poisoned or something. What I did was:

1. Uninstalled and reinstalled Anaconda (`rm -rf ~/anaconda`, remove conda init section from `.bashrc` and `.zshrc`, reinstall)
2. Changed default shell to bash with (`chsh -s /bin/bash` and again after `sudo -i`). Also changed path in terminal.
3. Uninstalled and reinstalled VSCode with a fresh download (deleted from Applications + https://stackoverflow.com/questions/42603103/how-to-completely-uninstall-vscode-on-mac).
4. Made sure preferences were empty.
5. Rebooted
6. Opened up integrated terminal + `which python`. Result was `/usr/bin/python`. Shell is `bash`.
7. In iTerm or Terminal it is `/Users/hameerabbasi/anaconda/bin/python`.
